waltz-client/src/main/java/com/wepay/waltz/client/internal/network/WaltzNetworkClient.java

```diff
@@ -222,6 +233,45 @@ public void requestHighWaterMark(ReqId reqId) {
         }
     }
 
+    /**
+     * Requests list of partitions assigned on that server
+     *
+     * @return Completable future that will container the list of partitions assigned on that server
+     * @throws InterruptedException If thread interrupted while waiting for channel to be ready
+     */
+    public Future<List<Integer>> getServerPartitionAssignments() throws InterruptedException {
+        synchronized (lock) {
+            if (running && !channelReady) {
+                lock.wait(CHANNEL_NOT_READY_TIMEOUT);
+            }
+
```

Comment:
might want to handle `if (!running)` condition as well after coming out of the lock.wait

Reply:
fixed. Thanks!